ksql-engine/src/main/java/io/confluent/ksql/function/InternalFunctionRegistry.java

```
@@ -200,15 +212,26 @@ private void init() {
 
   }
 
-  @Override
-  public KsqlFunction getFunction(final String functionName) {
+  public UdfFactory getUdfFactory(final String functionName) {
     return ksqlFunctionMap.get(functionName.toUpperCase());
   }
 
   @Override
   public boolean addFunction(final KsqlFunction ksqlFunction) {
     final String key = ksqlFunction.getFunctionName().toUpperCase();
-    return ksqlFunctionMap.putIfAbsent(key, ksqlFunction) == null;
+    try {
+      ksqlFunctionMap.compute(key, (s, udf) -> {
+        if (udf == null) {
+          udf = new UdfFactory(key, ksqlFunction.getKudfClass(), ksqlFunction.getReturnType());
+        }
+        udf.addFunction(ksqlFunction);
+        return udf;
+      });
+    } catch (KsqlException e) {
```

Why do we swallow the exception here? Would it not be safer to throw this up?